src/Task/Filesystem/CopyDir.php

```diff
@@ -113,14 +114,15 @@ protected function copyDir($src, $dst)
             mkdir($dst, $this->chmod, true);
         }
         while (false !== ($file = readdir($dir))) {
-            if (in_array($file, $this->exclude)) {
-                 continue;
+            // Support basename and full path exclusion.
+            if (in_array($file, $this->exclude) || in_array($parent . $file, $this->exclude) || in_array($src . DIRECTORY_SEPARATOR . $file, $this->exclude)) {
```

Comment:
Provided that we are talking about the command arguments as they are specified in test, `nested3/nested31/structu311.re` will be excluded because the whole directory `nested3/nested31` is already excluded.
But if you are referring to the case that only one relative directory can be excluded - this will work with more than one directory (test for this case in next commit).